mlrun/model_monitoring/db/stores/v3io_kv/kv_store.py

```diff
@@ -281,66 +345,63 @@ def write_application_event(
                       object.
         :param kind: The type of the event, can be either "result" or "metric".
         """
-        if kind == mm_constants.WriterEventKind.METRIC:
-            # TODO : Implement the logic for writing metrics to KV
-            return
-
-        endpoint_id = event.pop(mm_constants.WriterEvent.ENDPOINT_ID)
-        app_name = event.pop(mm_constants.WriterEvent.APPLICATION_NAME)
-        metric_name = event.pop(mm_constants.ResultData.RESULT_NAME)
-        attributes = {metric_name: json.dumps(event)}
 
-        v3io_monitoring_apps_container = self.get_v3io_monitoring_apps_container(
-            project_name=self.project
-        )
+        container = self.get_v3io_monitoring_apps_container(project_name=self.project)
+        endpoint_id = event.pop(mm_schemas.WriterEvent.ENDPOINT_ID)
+
+        if kind == mm_schemas.WriterEventKind.METRIC:
```

Comment:
just a recommendation, I think it's better to wrap all of this if clause in a different function and keep the `write_application_event` more "clean". I mean, adding something like
`table_path, key, attributes = self._generate_application_event_attributes(endpoint_id, event, kind)`...

Reply:
I agree with you that the implementation can be cleaner.
The main goal here was to do it fast.
How about deferring that? Once we think about all the cases, we may have a more holistic approach, not only for V3IO KV.